dev/faq/data-migration.md

```diff
@@ -23,17 +23,42 @@ The `driver: bad connection` error indicates that anomalies have occurred in the
 
 When this type of error occurs in the current version, use `stop-task` to stop the task and then use `start-task` to restart the task. The automatic retry mechanism of DM will be improved later.
 
-## What can I do when a replication task is interrupted with failing to get or parse binlog errors like `get binlog error ERROR 1236 (HY000)` and `binlog checksum mismatch, data may be corrupted` returned?
+## What can I do when the relay unit throws error `event from * in * diff from passed-in event *` or a replication task is interrupted with failing to get or parse binlog errors like `get binlog error ERROR 1236 (HY000)` and `binlog checksum mismatch, data may be corrupted` returned?
 
-During the DM incremental replication process, this error might occur if the binlog file in the upstream exceeds **4 GB**, and DM encounters a replication interruption when processing this binlog file (including interruption caused by anomalies in ordinary pause or stop tasks).
+During the DM process of relay log pulling or incremental replication, this two errors might occur if the binlog file in the upstream exceeds **4 GB**.
 
-**Cause:** DM needs to store the replicated binlog position, and MySQL officially uses `uint32` to store it, so the binlog position of the file with offset exceeding **4 GB** overflows and an incorrect binlog position is stored. After the task or DM-worker is restarted, this incorrect binlog position is used to re-parse the binlog or relay log.
+**Cause:** When writing relay logs, DM needs to perform event verification based on binlog positions and the size of the binlog file, and store the replicated binlog positions as checkpoints. However, the official method of MySQL to store binlog positions is `uint32`. This means the binlog position offset for a binlog file over 4 GB overflows, hence the errors above.
```

Comment:
`hence` is usually used as an adverb.

```suggestion
**Cause:** When writing relay logs, DM needs to perform event verification based on binlog positions and the size of the binlog file, and store the replicated binlog positions as checkpoints. However, the official MySQL uses `uint32` to store binlog positions. This means the offset value of binlog positions over 4 GB overflows and then the errors above occur.
```

Reply:
> binlog positions over 4 GB

Can you confirm @zimulala

Reply:
I think `This means the binlog position for a binlog file over 4 GB overflows, hence the errors above.` is fine. @dcalvin @lilin90

Reply:
Thanks for confirming